modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/DeployService.java

```diff
@@ -278,114 +348,87 @@ private List<DeployResourceEntity> getDeployResourceEntityList(
         }
+
         try {
-            deployServiceEntity.setServiceDeploymentState(ServiceDeploymentState.DESTROYING);
-            deployServiceStorage.storeAndFlush(deployServiceEntity);
-            String stateFile = deployServiceEntity.getPrivateProperties().get(STATE_FILE_NAME);
-            deployment.destroy(deployTask, stateFile);
-        } catch (Exception e) {
-            log.error("asyncDestroyService failed", e);
-            deployServiceEntity.setResultMessage(e.getMessage());
-            deployServiceEntity.setServiceDeploymentState(ServiceDeploymentState.DESTROY_FAILED);
-            DeployServiceEntity storedDeployServiceEntity =
-                    deployServiceStorage.storeAndFlush(deployServiceEntity);
-            if (Objects.nonNull(storedDeployServiceEntity)) {
-                deployment.deleteTaskWorkspace(deployTask.getId().toString());
+            DeployServiceEntity updatedDeployServiceEntity =
+                    flushDestroyServiceEntity(destroyResult, deployServiceEntity);
+            if (ServiceDeploymentState.DESTROY_SUCCESS
+                    == updatedDeployServiceEntity.getServiceDeploymentState()) {
+                deployment.deleteTaskWorkspace(destroyTask.getId().toString());
             }
+        } catch (RuntimeException e) {
+            log.info("Destroy service with id:{} update database entity failed. error:{}",
+                    destroyTask.getId(), e);
         }
-
     }
 
-    /**
-     * purge the service based on the serviceDeploymentState.
-     *
-     * @param deployment deployment
-     * @param deployTask deployTask
-     */
-    public void purgeService(Deployment deployment, DeployTask deployTask) {
-        DeployServiceEntity deployServiceEntity =
-                deployServiceStorage.findDeployServiceById(deployTask.getId());
-        ServiceDeploymentState serviceDeploymentState =
-                deployServiceEntity.getServiceDeploymentState();
-        if (serviceDeploymentState == ServiceDeploymentState.DEPLOY_FAILED
-                || serviceDeploymentState == ServiceDeploymentState.DESTROY_SUCCESS
-                || serviceDeploymentState == ServiceDeploymentState.MANUAL_CLEANUP_REQUIRED) {
-            asyncPurgeService(deployment, deployTask, deployServiceEntity);
-        } else {
-            throw new InvalidServiceStateException(
-                    String.format("Service %s is not in the state allowed for purging.",
-                            deployServiceEntity.getId()));
+    private String getStoredStateContent(DeployServiceEntity deployServiceEntity) {
+        if (Objects.isNull(deployServiceEntity)
+                || CollectionUtils.isEmpty(deployServiceEntity.getPrivateProperties())
+                || StringUtils.isEmpty(
+                deployServiceEntity.getPrivateProperties().get(STATE_FILE_NAME))) {
+            throw new IllegalStateException("Can't find valid state context in stored service.");
```

Comment:
Please throw any of our custom exception

Reply:
Updated.